logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/selfdeletingMessages/ObserveTeamSettingsSelfDeletingStatusUseCase.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.selfdeletingMessages
+
+import com.wire.kalium.logic.configuration.UserConfigRepository
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.logic.kaliumLogger
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlin.time.Duration
+
+/**
+ * Use case to observe the status of the self deletion settings set by the team admin.
+ */
+interface ObserveTeamSettingsSelfDeletingStatusUseCase {
```

Comment:
do we really need two use cases ? Is this something that can be fetched by the client in both cases ? Isn''t there the settings that would override other setting ? OR we could have situtaion where we need both use cases ?

Reply:
thanks for the clarification